### ISSUE TITLE
The target "CreateManifestResourceNames" does not exist in the project.	

### ISSUE THREAD

**Issue by TFTomSun** [microsoft/MSBuildSdks#120]
I get this error when I try to build this project:

> The target "CreateManifestResourceNames" does not exist in the project.	

```xml

<Project Sdk="Microsoft.Build.Traversal/2.0.12"  ToolsVersion="15.0">
  <ItemGroup>
    <!-- Build all projects recursively under the "src" folder -->
    <ProjectReference Include="src\**\*.*proj" />
  </ItemGroup>
</Project>


```

**Comment by TFTomSun** [comment]
if I add the target to the project I get the following error messages, but the build seems to work fine:

> NU1201: Project CrossCutting.Build.CommonOutput.Sdk is not compatible with net40 (.NETFramework,Version=v4.0). Project CrossCutting.Build.CommonOutput.Sdk supports: netstandard2.0 (.NETStandard,Version=v2.0)
> NU1201: Project CrossCutting.Build is not compatible with net40 (.NETFramework,Version=v4.0). Project CrossCutting.Build supports: netstandard2.0 (.NETStandard,Version=v2.0)
> NU1201: Project CrossCutting.Cli is not compatible with net40 (.NETFramework,Version=v4.0). Project CrossCutting.Cli supports: netcoreapp2.2 (.NETCoreApp,Version=v2.2)
> NU1201: Project CrossCutting.Nunit is not compatible with net40 (.NETFramework,Version=v4.0). Project CrossCutting.Nunit supports: netstandard1.1 (.NETStandard,Version=v1.1)
> NU1201: Project CrossCutting.Build.CommonOutput.Sdk is not compatible with net40 (.NETFramework,Version=v4.0) / win. Project CrossCutting.Build.CommonOutput.Sdk supports: netstandard2.0 (.NETStandard,Version=v2.0)
> NU

**Comment by jeffkl** [comment]
* Which version of MSBuild are you using? `msbuild /version`
* Which project reports that the target `CreateManifestResourceNames` is missing?
* Would you be able to set up a minimal repro that I can use to investigate?

Thanks!

**Comment by BasilAtMicrosoft** [comment]
I saw this exact issue and completely stumbled the correct solution (in my case) after unsuccessfully trying Tom's solution of adding an empty target.

In my case the team has a global package reference to a localization solution, say <PackageReference Include="Localization" … /> which included build\localization.targets.  The project throwing the CreateManifestResourceNames error had no localization logic.

The solution was to add <PackageReference Remove="Localization" /> to the unhappy project.  No more issues.

**Comment by jeffkl** [comment]
Closing old issues, please re-open with a minimal repro.

**Comment by ViktorHofer** [comment]
Hitting the same with msbuild 16.6 .NET Core in dotnet/runtime:

```
C:\git\runtime4\.dotnet\sdk\5.0.100-preview.3.20170.6\Microsoft.Common.CurrentVersion.targets(2795,7): error MSB4057: The target "CreateManifestResourceNames" does not exist in the project. [C:\git\runtime4\src\libraries\Native\native-binplace.proj]
```

**Comment by tebeco** [comment]
speaking in my own name (not from MSFT) but i suggest creating an open source Github reproduction repository

and as this issue is closed, you might want to open a new one, especially if you only have this with 5.x

**Comment by jeffkl** [comment]
@ViktorHofer can you please attach a preprocess log so I can see what's trying to run this target?

```cmd
dotnet msbuild /pp:pp.log
```

**Comment by Kieranties** [comment]
I've just met this issue and found it to occur when _not_ using `.csproj`

Failing:
```
  Restore completed in 35.68 ms for ...\packages\project.proj.
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 34.2 ms for ...\packages\project.proj.
C:\Program Files\dotnet\sdk\3.1.201\Microsoft.Common.CurrentVersion.targets(2795,7): error MSB4057: The target "CreateManifestResourceNames" does not exist in the project. [...\packages\project.proj]
```

Success:
```
  Restore completed in 2.71 sec for ...\packages\project.csproj.
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 41.49 ms for ...\packages\project.csproj.
  project -> ...\packages\bin\Debug\net452\project.dll
  project -> ...\packages\results\
```

**Comment by jeffkl** [comment]
I am not able to repro this at all.  I've attached what I'm using and it works just fine.

[example.zip](https://github.com/microsoft/MSBuildSdks/files/4446785/example.zip)

Can someone who repros it please do one of the following:
1. Zip up a set of projects that reproduces the error
2. Attach a binary log after reproducing the error when running `dotnet build /bl`
3. Attach a preprocess log by running `dotnet msbuild /pp:pp.log`

**Comment by sigurn** [comment]
Here is what reproduces the issue in my case:
[arm.zip](https://github.com/microsoft/MSBuildSdks/files/4673399/arm.zip)

**Comment by tclancy** [comment]
If it helps anyone else (as perverse as I am), I run into this off and on when building [Power Apps Components](https://docs.microsoft.com/en-us/powerapps/developer/component-framework/implementing-controls-using-typescript). Sometimes `msbuild /t:build` works but sometimes it fails with this message. I just had it happen again immediately after a successful build. All I had changed in the interim was some documentation. However, running `msbuild /t:build /restore` fixed it.

It should be said this almost certainly user error or being overly bold: I have customized the paths to fit what I want and I am running this 95% on a Mac with only the server and the build step (and, for added fun, installing `npm` packages) running in a Windows VM.
